index.d.ts

```diff
@@ -1,11 +1,13 @@
 import React from "react";
 import { SpringConfig } from "react-spring";
 
+type configsFn = ({numberValue, index}: {numberValue: number, index: number}) => SpringConfig;
```

Comment:
Possibly this would be better for backwards compatibility...
```suggestion
type configsFn = (numberValue: number, index: number) => SpringConfig;
```

Reply:
Good approach.

However, to change this, not only the type definition file be changed, but the actual code must also be changed and the documentation must be updated.

I would really appreciate it if you could work together.

Reply:
@heyman333, I'd be happy to further enhance this PR although I'm not clear on what you mean by your last comment.

> the documentation must be updated
The documentation is already written to support this [see here](https://github.com/heyman333/react-animated-numbers/blob/master/README.md?plain=1#L23-L24)

> the actual code must also be changed
Likewise, the code also supports this already too, [see here](https://github.com/heyman333/react-animated-numbers/blob/master/index.jsx#L38-L43).

--

From what I can see the only thing missing is the type definition which I belive is added by this PR. I would appreciate it if you could take another look at this one?

Reply:
hello @JohnGale87 
sorry for late reply.

[in your code](https://github.com/JohnGale87/react-animated-numbers/blob/fix-types/index.d.ts)
the type definition of  configsFn is as below
```ts
type configsFn = ({numberValue, index}: {numberValue: number, index: number}) => SpringConfig;
```
But actual code takes two consecutive arguments.
```js
  const setConfig = (configs, number, index) => {
    if (typeof configs === "function") {
      return configs(number, index);
    }
    return configs
      ? configs[getRandomIntInclusive(0, configs.length - 1)]
      : undefined;
  };

```

so you can try
1. Define configsFn to take two consecutive arguments again
2. Modify the actual code so that configsFn receives an Object type as an argument, and update the documentation.

Reply:
Ah I see. Yes, I think this is done now.